prog.go

```diff
@@ -685,31 +711,45 @@ func resolveBTFType(name string, progType ProgramType, attachType AttachType) (b
 		var target btf.Func
 		err := findKernelType("bpf_lsm_"+name, &target)
 		if errors.Is(err, btf.ErrNotFound) {
-			return nil, &internal.UnsupportedFeatureError{
+			return nil, -1, &internal.UnsupportedFeatureError{
 				Name: name + " LSM hook",
 			}
 		}
 		if err != nil {
-			return nil, fmt.Errorf("resolve BTF for LSM hook %s: %w", name, err)
+			return nil, -1, fmt.Errorf("resolve BTF for LSM hook %s: %w", name, err)
 		}
 
-		return &target, nil
+		return &target, -1, nil
 
 	case match{Tracing, AttachTraceIter}:
 		var target btf.Func
 		err := findKernelType("bpf_iter_"+name, &target)
 		if errors.Is(err, btf.ErrNotFound) {
-			return nil, &internal.UnsupportedFeatureError{
+			return nil, -1, &internal.UnsupportedFeatureError{
 				Name: name + " iterator",
 			}
 		}
 		if err != nil {
-			return nil, fmt.Errorf("resolve BTF for iterator %s: %w", name, err)
+			return nil, -1, fmt.Errorf("resolve BTF for iterator %s: %w", name, err)
+		}
+
+		return &target, -1, nil
+
+	case match{Extension, AttachNone}:
+		targetProg, err := NewProgramFromFD(attachTarget)
```

Comment:
This is just a bug in managing the resources. This should be closed inside this function.

Reply:
Actually, this is an issue with Close semantics, I think? If we close `targetProg` since it uses the same FD as the actual target program, it closes the (potentially only) FD to the program and can cause attachment to fail.
Maybe I'm not clear on ownership semantics, but I would have though that in general, calling `NewProgramFromFD()` would not take ownership as the FD must have come from elsewhere, and elsewhere would already be owning. If passing ownership were intended, then the `*Program`, not the FD would be passed?

Reply:
I'm adding a call to `Clone()` to try and make ownership a bit more clear, but maybe a comment would be better, instead?

Reply:
Also, maybe what's actually desired is `CloneProgramFromFD()` as a new helper?

Reply:
Of course, all of the above becomes moot if `targetProg` becomes a `*Program` (which I am looking into how to accomplish without breaking attaching to other object types).

Reply:
Addressed in 3499980